cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java

```diff
@@ -293,41 +310,73 @@ public final void storeTestEvent(StoredTestEvent event) throws IOException
+	 * Asynchronously writes to storage messages of test event
+	 * @param eventId ID of test event
 	 * @param batchId ID of batch where event is stored, if applicable
 	 * @param messagesIds collection of stored message IDs
 	 * @return future to get know if storing was successful
 	 */
 	public final CompletableFuture<Void> storeTestEventMessagesLinkAsync(StoredTestEventId eventId, StoredTestEventId batchId, Collection<StoredMessageId> messagesIds)
 	{
-		logger.debug("Storing links between test event {} and {} message(s) asynchronously", eventId, messagesIds.size());
+		logger.debug("Storing {} message(s) of test event {}", messagesIds.size(), eventId);
 		return doStoreTestEventMessagesLinkAsync(eventId, batchId, messagesIds)
 				.whenComplete((r, error) -> {
 					if (error != null)
-						logger.error("Error while storing links between test event "+eventId+" and "+messagesIds.size()+" message(s) asynchronously", error);
+						logger.error("Error while storing "+messagesIds.size()+" message(s) of test event "+eventId+" asynchronously", error);
 					else
-						logger.debug("Links between test event {} and {} message(s) have been stored asynchronously", eventId, messagesIds.size());
+						logger.debug("Message(s) {} of test event {} have been stored asynchronously", messagesIds.size(), eventId);
 				});
 		
 	}
-	
+
+	/**
+	 * Writes to storage test event of messages
+	 * @param eventId ID of stored test event
+	 * @param batchId ID of batch where event is stored, if applicable
+	 * @param messagesIds collection of message IDs
+	 * @throws IOException if data writing failed
+	 */
+	public final void storeMessageTestEventLink(StoredTestEventId eventId, StoredTestEventId batchId, Collection<StoredMessageId> messagesIds) throws IOException
+	{
+		logger.debug("Storing test event {} of {} message(s)", eventId, messagesIds.size());
+		doStoreMessageTestEventLink(eventId, batchId, messagesIds);
+		logger.debug("Test event {} of {} message(s) has been stored", eventId, messagesIds.size());
+	}
+
+	/**
+	 * Asynchronously writes to storage test event of messages
+	 * @param eventId ID of stored test event
+	 * @param batchId ID of batch where event is stored, if applicable
+	 * @param messagesIds collection of message IDs
+	 * @return future to get know if storing was successful
+	 */
+	public final CompletableFuture<Void> storeMessageTestEventLinkAsync(StoredTestEventId eventId, StoredTestEventId batchId, Collection<StoredMessageId> messagesIds)
+	{
+		logger.debug("Storing test event {} of {} message(s) asynchronously", eventId, messagesIds.size());
+		return doStoreMessageTestEventLinkAsync(eventId, batchId, messagesIds)
+				.whenComplete((r, error) -> {
+					if (error != null)
+						logger.error("Error while storing test event "+eventId+" of "+messagesIds.size()+" message(s) asynchronously", error);
+					else
+						logger.debug("Test event {} of {} message(s) has been stored asynchronously", eventId, messagesIds.size());
+				});
+
+	}

```

Comment:
Are these methods really needed? This is part of Cassandra-dependent implementation.